packages/playground/data-liberation/src/wxr/WP_WXR_Reader.php

```diff
@@ -866,6 +866,9 @@ private function after_entity() {
 	}
 
 	public function current(): object {
+		if ( null === $this->entity_data && ! $this->is_finished() && ! $this->get_last_error() ) {
+			$this->next();
```

Comment:
@adamziel What is the purpose of calling next() within current()?

From the outside, I would expect current() to have no side-effects. It seems like it could also lead to consumers unintentionally skipping a first entity if current() happens to be called before the consumers' first explicit next() call.

Reply:
That's for lazy initializing the iterator. Foreach() calls next() after every iteration but not before. We can either eagerly initialize the iterator in the constructor or in the first current() call. An inline comment to document that could be useful.